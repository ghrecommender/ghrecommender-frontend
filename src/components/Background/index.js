// @flow strict

import React from 'react'

import Particles from 'react-particles-js'

const color = '#5382C5'
const count = window.innerWidth / 25 / window.devicePixelRatio


const options = {
  particles: {
    number: {
      value: count,
      density: {
        enable: true,
        value_area: 800,
      },
    },
    color: {
      value: color,
    },
    shape: {
      type: 'circle',
      stroke: {
        width: 0,
        color: '#000000',
      },
      polygon: {
        nb_sides: 5,
      },
    },
    opacity: {
      value: 0.5,
      random: false,
      anim: {
        enable: false,
        speed: 1,
        opacity_min: 0.1,
        sync: false,
      },
    },
    size: {
      value: 5,
      random: true,
      anim: {
        enable: false,
        speed: 40,
        size_min: 0.1,
        sync: false,
      },
    },
    line_linked: {
      enable: true,
      distance: 150,
      color: color,
      opacity: 0.4,
      width: 1,
    },
    move: {
      enable: true,
      speed: 6,
      direction: 'none',
      random: false,
      straight: false,
      out_mode: 'bounce',
      bounce: false,
      attract: {
        enable: false,
        rotateX: 600,
        rotateY: 1200,
      },
    },
  },
  interactivity: {
    detect_on: 'window',
    onresize: {
      density_auto: true,
    },
    events: {
      onhover: {
        enable: true,
        mode: 'repulse',
      },
      onclick: {
        enable: true,
        mode: 'push',
      },
      resize: true,
    },
    modes: {
      grab: {
        distance: 400,
        line_linked: {
          opacity: 1,
        },
      },
      bubble: {
        distance: 400,
        size: 40,
        duration: 2,
        opacity: 8,
        speed: 3,
      },
      repulse: {
        distance: 200,
        duration: 0.4,
      },
      push: {
        particles_nb: 4,
      },
      remove: {
        particles_nb: 2,
      },
    },
  },
  retina_detect: true,
}

const Background = () => (
  <Particles
    style={{
      position: 'absolute',
      top: 0,
      left: 0,
      width: '100%',
      height: '100%',
    }}
    params={options}
  />
)

export default Background
