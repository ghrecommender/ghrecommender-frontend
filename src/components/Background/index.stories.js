import React from 'react'

import Background from './index'

const style = { backgroundColor: '#1b1c1d', width: '100%', height: '100%', position: 'absolute' }

export default {
  title: 'Components|Background',
  decorators: [story => <div style={style}>{story()}</div>],
  component: Background,
}

export const background = () => <Background />
