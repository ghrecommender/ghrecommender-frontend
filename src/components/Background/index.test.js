import React from 'react'
import rendered from 'react-test-renderer'

import Background from './index'


test('<Background />', () => {
  const component = rendered.create(<Background />)
  expect(component.toJSON()).toMatchSnapshot()
})
