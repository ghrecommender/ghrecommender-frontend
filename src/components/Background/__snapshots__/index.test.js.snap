// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> 1`] = `
<div>
  <canvas
    style={
      Object {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  />
</div>
`;
